### PR TITLE
Bindings for Vehicle Crash Detection APIs

### PR DESCRIPTION
**Pull request by countcain**
FYI this is still in progress.

### CODE REVIEW

File: ios/RNSentiance.m [hunk 1]
```diff
@@ -895,4 +932,25 @@ - (NSDictionary*)convertTripProfileToDict:(SENTTripProcessingTripProfile*) tripP
...
+- (NSDictionary*)convertVehicleCrashEventToDict:(SENTVehicleCrashEvent*) crashEvent {
+    NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
+    double time = [crashEvent.date timeIntervalSince1970] * 1000;
+    dict[@"time"] = @(time);
+
+
+    if(crashEvent.location != nil) {
+        NSDictionary *location = @{
+                                   @"latitude": @(lastKnownLocation.coordinate.latitude),
+                                   @"longitude": @(lastKnownLocation.coordinate.longitude)
```
[the remark this answers is not in the document]
Comment (countcain) [reply ios/RNSentiance.m]: Corrected! it is copied from the last deprecated crash event handler. 😅

File: ios/SENTSDK.podspec [hunk 2]
```diff
@@ -1,13 +1,13 @@
 Pod::Spec.new do |s|
     s.name              = 'SENTSDK'
-    s.version           = '5.7.4'
+    s.version           = '5.9.0'
```
Comment (sebouh00) [inline ios/SENTSDK.podspec:3]: The changes in this file should point to `5.8.0` right?

Comment (sebouh00) [reply ios/SENTSDK.podspec:3]: Having said that, `5.8.0` and `4.17.0` will be skipped. So this version is technically what we eventually want (perhaps in a follow-up PR?), but it doesn't align with the description in the changelog.

Comment (countcain) [reply ios/SENTSDK.podspec:3]: I prefer we directly pointing it to 5.9.0 as there was no actual 5.8.0 public release. The changelog is updated.

Comment (sebouh00) [reply ios/SENTSDK.podspec:3]: Sounds good. Note that `5.9.0` will include `invokeDummyVehicleCrash` and `isVehicleCrashDetectionSupported` implementations. So perhaps this change should include those RN implementations as well.

Comment (countcain) [reply ios/SENTSDK.podspec:3]: Yes. Once those two new methods are ready. I will implement the corresponding bindings.

**Comment by prakashraman** [comment]
hey @greenzeal - have updated the implementation of the 2 missing methods. Do review. 

Open questions:
- Should we add a button in the example app to "invoke" crash
- Any other change in the UI
- Should we update anything else in the README/Changelog /

File: ios/RNSentiance.m [hunk 3]
```diff
@@ -2,7 +2,7 @@
 #import <SENTSDK/SENTSDK.h>
 #import <SENTSDK/SENTSDKStatus.h>
 #import <SENTSDK/SENTPublicDefinitions.h>
-#import <RNSentianceNativeInitialization.h>
+#import "RNSentianceNativeInitialization.h"
```
Comment (countcain) [inline ios/RNSentiance.m:5]: 👍🏻

**Comment by countcain** [comment]
> * Should we add a button in the example app to "invoke" crash

I suggest we demonstrate all three methods in the example app crash-detection branch. The current crash detection listener in that branch is deprecated.

1. setup the new vehicleCrashDetection listener.
1. Add an `invoke` button.
2. check if vehicle crash detection is supported, if yes, add an `invoke` button. Clicking the button will trigger the listener and output the object to the screen as what currently in place.

> * Any other change in the UI

not really I can think of.

> * Should we update anything else in the README/Changelog?

I've updated the readme and changelog in previous commits. In the changelog, the unreleased portion will need to be updated to the actual release version once we release it.

**Review by sebouh00 — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- Upgrade iOS SDK to [5.9.0]()
+- Upgrade Android SDK to [4.18.0]()
```
Comment (countcain) [inline CHANGELOG.md:10]: https://docs.sentiance.com/sdk/changelog/android#4-18-0-14-jan-2021 @prakashraman can you help add the link here? thx.

File: ios/RNSentiance.m [hunk 5]
```diff
@@ -605,6 +605,46 @@ - (NSString *) getValueForKey:(NSString *)key value:(NSString *)defaultValue {
...
+        }];
+        resolve(@(YES));
+    } @catch (NSException *e) {
+        reject(e.name, e.reason, nil);
+    }
+}
+
+RCT_EXPORT_METHOD(invokeDummyVehicleCrash:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [[SENTSDK sharedInstance] invokeDummyVehicleCrashHandler];
```
Comment (sebouh00) [inline ios/RNSentiance.m:627]: Note that the iOS implementation changed this method name to `invokeDummyVehicleCrash`.

**Pull request merged**